### ISSUE TITLE
Precision is 1:0.0041

### ISSUE THREAD

**Issue by MTime90** [ayush29feb/Sketch-A-XNORNet#3]
Hi, ayush.
I have trained this model for several days. But what I got just is: 

Running Evaluation on testset
Num examples: 6500  Num correct: 26  Precision @ 1: 0.0040 (3479.218 sec)
Running Evaluation on trainset
 Num examples: 13500  Num correct: 55  Precision @ 1: 0.0041 (1058.892 sec)

Could you figure me out where I did wrang? 
Thanks

**Comment by ayush29feb** [comment]
Hi, I apologize for the delayed response!

Can you tell me what model are you training? I used the pertained model provided by Sketch-A-Net and then applied the binary ops. The hyper-parameters are not tuned to start from scratch.

**Comment by SlideLucask** [comment]
Hi, ayush.

I used your binary ops to train the VGG16 model on the MNIST dataset . Have you done like this ?

I am not good at coding in tensorflow framwork . but in theory，it may not damege the accuracy poorly?

but i got only 10% acc ? it should be what i did wrong in the coding the vgg16 by binary ops or the binary ops did damage the accuracy ?  

Could you give some advice?

**Comment by ayush29feb** [comment]
Hi @SlideLucask, I haven't played with the VGG16 Model. But since this we are essentially converting a huge tensor with many float values to binary values with a single float values (the alpha) it definitely will loose precision. This is a trade-off optimization and precision.

However, I think you can definitely get better accuracy than 10%. I am not sure what is the best accuracy you'd get on VGG with binary ops but you can look at the implementation of original authors of XNOR-Net. I think they released AlexNet and ResNet versions at least. Also, tensorflow is not a good framework to use here because it doesn't support binary data types. The original authors also wrote a framework in C called darknet which has these binary ops. You may wanna try to use that instead. I was just trying to make a proof of concept here.
